src/internal_modules/roc_fec/reader.cpp

```diff
@@ -139,6 +141,13 @@ status::StatusCode Reader::get_next_packet_(packet::PacketPtr& ptr) {
     fill_block_();
 
     packet::PacketPtr pp = source_block_[next_packet_];
+    if (next_packet_ == 0) {
+        if (pp) {
+            update_block_duration_(pp);
+        } else {
+            prev_block_timestamp_valid_ = false;
+        }
+    }
```

Comment:
So this is triggered when the very first packet of a block is received.

What about moving this logic to `next_block_()`?

It is triggered when we're going to switch to the next block.

Inside `next_block_()`, even if the first packet was lost, there is the maximum chance that it was repaired.